nilearn/glm/tests/test_first_level.py

```diff
@@ -630,3 +630,22 @@ def test_first_level_predictions_r_square():
 
     r_square_2d = model.masker_.transform(r_square_3d)
     assert_array_less(0., r_square_2d)
+
+
+def test_warning_fit():
```

Comment:
I think this is too complicated for a simple warning check. I would be in favor of just checking this in `test_fmri_inputs` for example by adding something like:

```python
with pytest.warns(UserWarning,
                  match="If the design matrices are given"):
        FirstLevelModel(mask_img=mask).fit(fi, 
                                           design_matrices=d,
                                           confounds=conf)
```
WDYT?